shell/platform/android/io/flutter/app/FlutterPluginRegistry.java

```diff
@@ -110,24 +126,47 @@ public Context activeContext() {
 
         @Override
         public BinaryMessenger messenger() {
-            return mNativeView;
+            // NOTE: We have to decide what universe we're in before we can take action.
```

Comment:
nit: this comment doesn't add anything that isn't already expressed by the code.

Reply:
I opted to introduce an `isOriginalEmbedding()` method to make it completely clear what `mNativeView != null` actually means.